salt/grains/core.py

```diff
@@ -1657,13 +1678,20 @@ def os_data():
                                 codename = codename_match.group(1)
                         grains['lsb_distrib_codename'] = codename
                     if 'CPE_NAME' in os_release:
-                        if ":suse:" in os_release['CPE_NAME'] or ":opensuse:" in os_release['CPE_NAME']:
+                        cpe = _parse_cpe_name(os_release['CPE_NAME'])
+                        if not cpe:
+                            log.error('Broken CPE_NAME format in /etc/os-release!')
+                        elif cpe.get('vendor', '').lower() in ['suse', 'opensuse']:
                             grains['os'] = "SUSE"
                             # openSUSE `osfullname` grain normalization
                             if os_release.get("NAME") == "openSUSE Leap":
                                 grains['osfullname'] = "Leap"
                             elif os_release.get("VERSION") == "Tumbleweed":
                                 grains['osfullname'] = os_release["VERSION"]
+                        # Override VERSION_ID, if CPE_NAME around
+                        if cpe.get('version'):
+                            grains['lsb_distrib_release'] = cpe['version']
```

Comment:
IIUC this would make the `lsb_distrib_release` grain to change from current `12.3` (VERSION_ID) to `sp3` (phase on CPE) on SLE12SP3. Is that correct?

In that case, I wonder if we want to make this change a part of a bugfix release.

Reply:
Fixed.